heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/service/OAuthInterceptorService.java

```diff
@@ -400,11 +397,18 @@ private void generateResponseWithSuccess(String message) {
      * OAuth2.0 Password Flow
      */
 
-    private void passwordFlow(Provider provider, OAuthRequest oAuthRequest, String clientId, String privateKey, int timeAccessToken, int timeRefreshToken, String claimsJson, String accessToken) {
-        validateClientId(clientId);
-        validateInProvider(provider, clientId, accessToken);
+    private void passwordFlow(Provider provider, OAuthRequest oAuthRequest, HttpServletRequest request, String privateKey, int timeAccessToken, int timeRefreshToken) {
+        String body = "";
 
-        TokenOAuth tokenOAuth = oAuthService.generateTokenOAuth(oAuthRequest, oAuthRequest.getClientId(), privateKey, timeAccessToken, timeRefreshToken, claimsJson);
+        try {
```

Comment:
the code from line 403 to 406 repeats 5 times. Consider extract it to a separated private method